src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -636,4 +629,42 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
             }
         }
     }
+
+    private getArchitecture(
```

Comment:
this kind of logic normally lives in `src/lambda/local/debugConfiguration.ts`, see for example `getCodeRoot`, `getHandlerName`, `getTemplate` ...